test/libraries/Hooks.t.sol

```diff
@@ -741,10 +746,13 @@ contract HooksTest is Test, Deployers, GasSnapshot {
         assertTrue(hookAddr.hasPermission(Hooks.AFTER_DONATE_FLAG));
     }
 
-    function test_validateHookAddress_failsAllHooks(uint152 addr, uint8 mask) public {
-        uint160 preAddr = uint160(uint256(addr));
-        vm.assume(mask != 0xff8);
-        IHooks hookAddr = IHooks(address(uint160(preAddr) | (uint160(mask) << 151)));
+    function test_validateHookAddress_failsAllHooks(uint160 addr, uint16 mask) public {
+        uint160 preAddr = uint160(uint256(addr) & clearAllHookPermisssionsMask);
```

Comment:
feels weird to go 160->256->160. Can we keep this always in a 256?

Reply:
do you mean keep always in a uint160?  I think we want to work with addresses (uint160)

Reply:
changed so that clearAllHookPermisssionsMask is uint160